bib_rights/gen_bib_rights_tests.pl

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/perl
+
+# This script is used to generate the fixtures used by t/bib_rights.t.
+# It need not be re-run unless coverage years or catalog records are added
+# to the test suite. Depends on HathiTrust Bib API for cached MARC record fixtures.
+# Creates a JSON file with the results of running the bib rights algorithm
+# on a selection of representative catalog records. The results are to be considered
+# authoritative for subsequent testing purposes.
+
+use strict;
+use warnings;
+BEGIN {
+  unshift(@INC, $ENV{'SDRROOT'}. '/crms/cgi');
+  unshift(@INC, $ENV{'SDRROOT'}. '/crms/post_zephir_processing');
+}
+
+use utf8;
+binmode(STDOUT, ':encoding(UTF-8)');
+use CRMS;
+use Getopt::Long qw(:config no_ignore_case bundling);
+use Encode;
+use Term::ANSIColor qw(:constants colored);
+$Term::ANSIColor::AUTORESET = 1;
+use Data::Dumper;
+use JSON::XS;
+use bib_rights;
+use MARC::File::XML(BinaryEncoding => 'utf8');
+
+my $crms = CRMS->new();
+my @years = (2010, 2015, 2020, 2025, 2030);
+my $tests = [
+  ['mdp.39015086492272', 'CHECK_GOV NIST-NSRDS'],
+  ['umn.31951d02932305y', 'CHECK_GOV NTIS'],
+  ['mdp.39015062147791', 'CHECK_GOV armed forces comm assoc'],
+  ['uc1.31822016975690', 'CHECK_GOV armed forces comm assoc (possible bug in RE?)'],
+  ['chi.16700319', 'CHECK_GOV national research council'],
+  # There is no "national research council canada" volume that is marked US fed doc.
+  #select * from bib_rights_bi where f008 regexp '^.{28}f' and imprint RLIKE 'national.research.council.+canada' and f008 regexp '^.{17}u';
+  #select * from bib_rights_bi where f008 regexp '^.{28}f' and imprint RLIKE 'canada.*national.research.council' and f008 regexp '^.{17}u';
+  
+  ['uc1.31822009441387', 'CHECK_GOV Smithsonian'],
+  ['umn.319510028195022', 'CHECK_GOV national gallery of art'],
+  ['njp.32101066807213', 'CHECK_GOV Federal Reserve'],
+  
+  ['njp.32101077883609', 'clean_pub_place nonalpha-space'],
+  ['mdp.39015053274380', 'clean_pub_place pr -> pru (Puerto Rico)'],
+  ['uc1.31158001472769', 'clean_pub_place us -> xxu'],
+  
+  
+  ['mdp.39015015147864', 'clean_date with 008 date1 0000'],
+  ['mdp.39015063938933', 'clean_date with 008 date2 0000'],
+  ['mdp.39015054190924', 'getBibFmt CF'],
+  ['mdp.39015011016204', 'getBibFmt VM'],
+  ['mdp.39015010377565', 'getBibFmt MU'],
+  ['uc1.32106000718277', 'getBibFmt MP'],
+  ['mdp.39015006077963', 'getBibFmt SE'],
+  ['ien.35558002752380', 'getBibFmt MX'],
+  ['mdp.39015006077963', 'getBibFmt SE from bibLev "s"'],
```

Comment:
Either bibLev "s" test is ineffective (bled off by an earlier test in the sub would be my guess) or the "uncoverable" comment is wrong.